pkg/cachemanager/cachemanager.go

```diff
@@ -130,22 +129,23 @@ func (c *CacheManager) UpsertSource(ctx context.Context, sourceKey aggregator.Ke
 	// in the manageCache loop.
 
 	err := c.replaceWatchSet(ctx)
-	if general, failedGVKs, danglingWatches := interpretErr(err, newGVKs); len(failedGVKs) > 0 || general {
-		var gvksToTryCancel []schema.GroupVersionKind
-		if general {
-			// if the err is general, assume all gvks need TryCancel because of some
-			// WatchManager internal error and we don't want to block readiness.
-			gvksToTryCancel = c.gvksToSync.GVKs()
-		} else {
-			gvksToTryCancel = failedGVKs
-
-			c.handleDanglingWatches(danglingWatches)
-		}
+	general, addGVKFailures, removeGVKFailures := interpretErr(err, newGVKs)
+	var gvksToTryCancel []schema.GroupVersionKind
+	if general {
+		// if the err is general, assume all gvks need TryCancel because of some
+		// WatchManager internal error and we don't want to block readiness.
+		gvksToTryCancel = c.gvksToSync.GVKs()
+	} else {
+		gvksToTryCancel = addGVKFailures
 
-		for _, g := range gvksToTryCancel {
-			c.tracker.TryCancelData(g)
-		}
+		c.handleDanglingWatches(removeGVKFailures)
```

Comment:
interpreting errors for failure to remove GVKs and adding them to gvksToDeleteFromCache can be integrated into `replaceWatchSet()` to avoid copy/paste. `interpretErr()` should not be involved.

This avoids complicating interpretErr()'s interface. It will also solve a bug in the current iteration, where a general error leads to no wipes/retries.

It also allows us to treat all the g